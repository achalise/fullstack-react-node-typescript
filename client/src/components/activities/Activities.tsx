import React from 'react';
import { IActivity } from '../../domain/activity';
import { Link } from 'react-router-dom';

export const Activities: React.FC<ActivitiesProps> = (props) => {
  let activities: IActivity[] = [
    {
      id: '1',
      name: `test activity`,
      type: `test type`,
      created: new Date(),
    },
  ];
  const noResults = <div>No Activities found</div>;
  return activities ? (
    <>
      <div className="d-flex justify-content-between">
        <div className="d-flex"><h1>Activities</h1></div>
        <div className="d-flex align-items-center"><Link to='/activity'> <button className="btn-sm btn-primary" >Add</button></Link></div>
      </div>
      <table className="table">
        <thead className="thead-dark">
          <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Type</th>
            <th scope="col">Created</th>
          </tr>
        </thead>
        <tbody>
          {activities.map((activity) => 
             (<tr key={activity.id}>
              <th scope="row">1</th>
              <td>{activity.name}</td>
              <td>{activity.type}</td>
              <td>{activity.created.toDateString()}</td>
            </tr>)
          )}
        </tbody>
      </table>
    </>
  ) : (
    noResults
  );
};
interface ActivitiesProps {}
