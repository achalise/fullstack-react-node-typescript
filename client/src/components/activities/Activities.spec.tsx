import React from 'react';
import {render, cleanup, waitForElement, wait} from '@testing-library/react'
import {Activities} from './Activities';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

describe(`Activities tests`, () => {
    it(`Should display the list of activities`, async (done) => {
        const {container, getByText} = render(<BrowserRouter><Activities/></BrowserRouter> );
        const element = await waitForElement(() => getByText('Activities'));
        expect(element).toMatchSnapshot();
        done();
    })
});
