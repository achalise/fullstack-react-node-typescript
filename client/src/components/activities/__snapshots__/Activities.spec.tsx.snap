// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activities tests Should display the list of activities 1`] = `
<h1>
  Activities
</h1>
`;
